src/flowmanager.js

```diff
@@ -109,20 +111,41 @@ export default class FlowManager extends EventEmitter {
       chunkCount = chunksToBeUploaded.length;
     } else {
       whiteListedFileEntries.map((file, index) => {
-        this.emit(events.FILE_UPLOAD_INITIATED, {
-          filename: file.name,
-          path: targetFolder.id,
-          taskId: tempIds[index],
-          data: file._data
-        });
-        this.emitUploader(internalEvents.UPLOAD_INITIATED, {
-          filename: file.name,
-          size: file.size,
-          progress: 0,
-          isComplete: false,
-          isError: false,
-          taskId: tempIds[index]
+        if (file.size) {
+          this.emit(events.FILE_UPLOAD_INITIATED, {
+            filename: file.name,
+            size: getHumanFileSize(file.size),
+            path: targetFolder.id,
+            taskId: tempIds[index],
+            data: file._data
+          });
+          this.emitUploader(internalEvents.UPLOAD_INITIATED, {
+            filename: file.name,
+            size: getHumanFileSize(file.size),
+            progress: 0,
+            isComplete: false,
+            isError: false,
+            taskId: tempIds[index]
+          });
+        } else {
+          file.file((fl) => {
+          this.emit(events.FILE_UPLOAD_INITIATED, {
+            filename: file.name,
+            size: getHumanFileSize(fl.size),
+            path: targetFolder.id,
+            taskId: tempIds[index],
+            data: file._data
+          });
+          this.emitUploader(internalEvents.UPLOAD_INITIATED, {
+            filename: file.name,
+            size: getHumanFileSize(fl.size),
+            progress: 0,
+            isComplete: false,
+            isError: false,
+            taskId: tempIds[index]
+          });
```

Comment:
Repeated code.. Make it a utility function and use it both in the if condition and the loop in else

Reply:
Created a utility to get data object.